mobile/library/common/jni/jni_interface.cc

```diff
@@ -832,8 +833,8 @@ static void* jvm_on_send_window_available(envoy_stream_intel stream_intel, void*
 
   jlongArray j_stream_intel = Envoy::JNI::native_stream_intel_to_array(jni_helper, stream_intel);
 
-  jobject result = Envoy::JNI::callObjectMethod(jni_helper, j_context, jmid_onSendWindowAvailable,
-                                                j_stream_intel);
+  jobject result =
+      jni_helper.callObjectMethod(j_context, jmid_onSendWindowAvailable, j_stream_intel).release();
```

Comment:
silly question perhaps: why do some of these JniHelper calls call `release()` on the result and some don't?

Reply:
Not silly at all :) This is for the case where I'm not sure of the lifetime of the reference. All those things where I have to use `release()` are the references passed into the callback just to keep the existing behavior the same. [std::unique_ptr<T>::release()](https://en.cppreference.com/w/cpp/memory/unique_ptr/release) basically relinquishes the ownership of the managed pointer. We may need to revisit this again in the future, but for now my goal is to not make any risky changes, so anything that I'm unsure about, I'll keep the behavior the same as the existing code.

Reply:
that makes perfect sense, thanks for the explanation!